source/slang/slang-ast-decl-ref.cpp

```diff
@@ -333,29 +333,54 @@ void DeclRefBase::toText(StringBuilder& out)
 
     SubstitutionSet substSet(this);
 
-    List<Decl*> decls;
-    for (auto dd = getDecl(); dd; dd = dd->parentDecl)
+    // Build a list of parent DeclRefs instead of just Decls
+    List<DeclRefBase*> declRefs;
+
+    for (DeclRefBase* dr = this; dr; dr = dr->getParent())
     {
+        auto dd = dr->getDecl();
+
+        // If this declaration is inside an extension, add it and then stop gathering parents
+        if (dd->parentDecl && as<ExtensionDecl>(dd->parentDecl))
+        {
+            declRefs.add(dr);
+            break; // Stop gathering parent DeclRefs to exclude namespace
+        }
+
         // Skip the module, file & include decls since their names are
         // considered "transparent"
-        //
         if (as<ModuleDecl>(dd) || as<FileDecl>(dd) || as<IncludeDecl>(dd))
             continue;
 
         // Skip base decls in generic containers. We will handle them when we handle the generic
         // decl.
-        //
         if (dd->parentDecl && as<GenericDecl>(dd->parentDecl))
             continue;
 
-        decls.add(dd);
+        declRefs.add(dr);
     }
 
-    decls.reverse();
+    declRefs.reverse();
 
     bool first = true;
-    for (auto decl : decls)
+    for (auto declRef : declRefs)
     {
+        auto decl = declRef->getDecl();
```

Comment:
In this loop you are already iterating through all parent decls, why checking for decl->parentDecl again?

Just check if decl is a extension decl and apply the logic there.